resources/indexer.py

```diff
@@ -1,13 +1,227 @@
-#!/usr/bin/env python
+# indexer.py
+"""Data-indexing script for File Catalog."""
+
 import argparse
-from concurrent.futures import ProcessPoolExecutor
+import asyncio
 import hashlib
-import json
 import os
-import requests
-import sys
+import re
+import tarfile
+from concurrent.futures import ProcessPoolExecutor
+from datetime import date
 from time import sleep
 
+import requests
+import xmltodict
+import yaml
+from icecube import dataclasses, dataio
+from rest_tools.client import RestClient
+
+SEASONS = {
+    '2005': 'ICstring9',
+    '2006': 'IC9',
+    '2007': 'IC22',
+    '2008': 'IC40',
+    '2009': 'IC59',
+    '2010': 'IC79',
+    '2011': 'IC86-1',
+    '2012': 'IC86-2',
+    '2013': 'IC86-3',
+    '2014': 'IC86-4',
+    '2015': 'IC86-5',
+    '2016': 'IC86-6',
+    '2017': 'IC86-7',
+    '2018': 'IC86-8',
+    '2019': 'IC86-9',
+    '2020': 'IC86-10'
+}
+
+
+def _get_year_from_season(season):
```

Comment:
Is there a reason for the leading underscore on function names? Just wondering.

Reply:
I use that to denote private functions. In this case there isn't much of a difference between public and private, but it's a habit.